### PR TITLE
Issue #2256 - Reinstate the ability to pull SRC files

### LINKED ISSUE #2256: Changes required for Unreal support
This is mostly a sandbox issue to add any endpoint required for @sebjf work with unreal implementation

- [x] Revive GLTF endpoint

### PR DESCRIPTION
**Pull request by sebjf**
This fixes #2256 

#### Description
Adds endpoints to return src files from the stash, and json objects to work with these containing information extracted from the history and scene.

### CODE REVIEW

**Review by carmenfan — COMMENTED** [review]
@sebjf  you don't need the GLTF stuff anymore right? I'll remove it if so

**Comment by sebjf** [comment]
@carmenfan No, we don't need the GLTF stuff. It seems travis has a few issues with my code formatting, so I can remove it if you like when I fix it?

**Comment by carmenfan** [comment]
> @carmenfan No, we don't need the GLTF stuff. It seems travis has a few issues with my code formatting, so I can remove it if you like when I fix it?

It's ok, don't worry about it. I can patch it up. I also want to refresh some of the promises to async/await for readability (Not your fault, the code you copied from is a bit dated 😆)

for linter stuff it's mostly just running a `yarn lint:fix` so it's pretty simple

File: backend/routes/model.js [hunk 1]
```diff
@@ -379,6 +380,91 @@ router.get("/:model/:uid.json.mpc",  middlewares.hasReadAccessToModel, getJsonMp
...
+ * @apiParam {String} teamspace Name of teamspace
+ * @apiParam {String} model The model Id to get unity assets for.
+ * @apiParam {String} rev The revision of the model to get src assets for
+ *
+ * @apiExample {get} Example usage:
+ * GET /teamSpace1/3549ddf6-885d-4977-87f1-eeac43a0e818/revision/07314a08-75da-46bf-9d2c-b7b20109143b/srcAssets.json HTTP/1.1
+ *
+ * @apiSuccessExample {json} Success:
+ * {
+ *   "models": [
```
Comment (carmenfan) [inline backend/routes/model.js:405]: @sebjf I got a couple questions:

1. How are you getting the model offset at the moment? The unityAsset.json version has the offset listed in here, from this example it doesn't look like you're extracting that? I looked into the Unreal code and it wasn't immedately clear where it's set.
2. Can we change the schema of this return? ideally this should be more inline with unityAssets, atm there's a lot of redundant info (i'll change it on the web side, just making sure you're happy with it on the unreal code change):
````json
{ "models": [
    {
         "database" : "sebjf",
          "model" : "9b458610-0c94-11eb-a083-2b878a4dc41e",
          "srcIDs" : [ "153cf665-2c84-4ff9-a9e2-ba495af8e6dc", "2967230f-67fa-45dc-9686-161e45c7c8a2", "07c67b6c-4b02-435f-8639-ea88403c36f7"]
    }
]}
````

**Comment by sebjf** [comment]
Hi @carmenfan, as far as I could tell, the Unity Assets API returns a document from the database containing the offset. This is created by the bouncer which determines the offset when it loads a scene. The SRC exporter doesn't have an equivalent document, so the SRC Assets API recovers the offset from the revision.
Yes please make any changes you like to the format, I can update the Unreal side easily enough!

**Comment by sebjf** [comment]
@carmenfan Just to confirm that these changes are all working in Unreal, and the updates to Unreal to use them have been pushed.

**Comment by carmenfan** [comment]
@sebjf the SRC generator also generates pairing json files: 
image

I just noticed you never wrote an endpoint for them, do you not need them? from what I rememebr that's where you get the original mesh info, material etc

**Comment by sebjf** [comment]
> @sebjf the SRC generator also generates pairing json files:
> image
> 
> I just noticed you never wrote an endpoint for them, do you not need them? from what I rememebr that's where you get the original mesh info, material etc

Hi @carmenfan, yes they are needed, but they are returned via FileRef class, called by the JSONAssets.getSuperMeshMapping method.

**Comment by carmenfan** [comment]
> > @sebjf the SRC generator also generates pairing json files:
> > image
> > I just noticed you never wrote an endpoint for them, do you not need them? from what I rememebr that's where you get the original mesh info, material etc
> 
> Hi @carmenfan, yes they are needed, but they are returned via FileRef class, called by the JSONAssets.getSuperMeshMapping method.

ahh of course we have *_unity.json.mpc, so it's just using that endpoint 👍

**Comment by carmenfan** [comment]
@sebjf I've made a tiny change in the logic for error catching for federations

can you pull the latest and confirm that it still works for you?

**Review by carmenfan — APPROVED** [review]

**Pull request merged**